### PR TITLE
[RHBPMS-4634] handle pom type dependencies in kie-ci

### PR DESCRIPTION
**Pull request by mariofusco**

### CODE REVIEW

File: drools-compiler/src/main/java/org/drools/compiler/kproject/ReleaseIdImpl.java [hunk 1]
```diff
@@ -78,7 +82,9 @@ public void setVersion(String version) {
 
     @Override
     public String toString() {
-        return groupId + ":" + artifactId + ":" + version;
+        return type == null || type.equals( "jar" ) ?
```
Comment (tarilabs) [inline drools-compiler/src/main/java/org/drools/compiler/kproject/ReleaseIdImpl.java:85]: looks ok

File: drools-compiler/src/main/java/org/drools/compiler/kproject/ReleaseIdImpl.java [hunk 2]
```diff
@@ -44,12 +45,15 @@ public ReleaseIdImpl(String releaseId) {
         this.version = split[2];
     }
 
-    public ReleaseIdImpl(String groupId,
-            String artifactId,
-            String version) {
+    public ReleaseIdImpl(String groupId, String artifactId, String version) {
+        this(groupId, artifactId, version, "jar");
```
Comment (tarilabs) [inline drools-compiler/src/main/java/org/drools/compiler/kproject/ReleaseIdImpl.java:49]: looks ok default jar.

File: drools-compiler/src/main/java/org/drools/compiler/kproject/ReleaseIdImpl.java [hunk 3]
```diff
@@ -142,28 +148,28 @@ private static ReleaseId getReleaseIdFromProperties(Properties props, String pat
...
-        if (artifactId != null ? !artifactId.equals(that.artifactId) : that.artifactId != null)
-            return false;
-        if (groupId != null ? !groupId.equals(that.groupId) : that.groupId != null)
-            return false;
-        if (version != null ? !version.equals(that.version) : that.version != null)
-            return false;
-
-        return true;
+    private boolean equalsIgnoreNull(Object o1, Object o2) {
+        return o1 != null ? o1.equals(o2) : o2 != null;
```
Comment (tarilabs) [inline drools-compiler/src/main/java/org/drools/compiler/kproject/ReleaseIdImpl.java]: If the o1 is diff than null, o1 must equals o2.
If the o1 is null, return "o2 is _diff than_ null" ? I think this second part is wrong?

**Review by tarilabs — CHANGES_REQUESTED** [review]
Can you kindly check `equalsIgnoreNull` comment, please?

File: drools-compiler/src/main/java/org/drools/compiler/kproject/ReleaseIdImpl.java (hunk 3 above)
Comment (mariofusco) [reply drools-compiler/src/main/java/org/drools/compiler/kproject/ReleaseIdImpl.java]: @tarilabs good catch, thanks

File: kie-ci/src/test/java/org/kie/scanner/KieModuleBuilderTest.java [hunk 4]
```diff
@@ -119,4 +138,74 @@ protected String createDRLWithImport( String ruleName ) {
...
+        File pomFile = fileManager.newFile("pom.xml");
+        fileManager.write(pomFile, pom);
+
+        InternalKieModule kJar1 = createKieJarWithPomDependencies( ks, releaseIdNoDep, pom );
+        MavenRepository repository = getMavenRepository();
+        repository.installArtifact(releaseIdNoDep, kJar1, pomFile);
+
+        InternalKieModule kJar2 = createKieJarWithDependencies( ks, releaseIdWithDep, true, "rule1", releaseIdNoDep);
+        KieContainer kieContainer2 = ks.newKieContainer( releaseIdWithDep );
+        KieSession kieSession = kieContainer2.newKieSession();
```
Comment (tarilabs) [inline kie-ci/src/test/java/org/kie/scanner/KieModuleBuilderTest.java:161]: I assume original problem was throwing Exception, hence the test does not need any assertions.

**Review by tarilabs — APPROVED** [review]
+1

File: kie-ci/src/test/java/org/kie/scanner/KieModuleBuilderTest.java (hunk 4 above)
Comment (mariofusco) [reply kie-ci/src/test/java/org/kie/scanner/KieModuleBuilderTest.java:161]: Yes, not sure what I could assert here. Maybe that ksession is not null?

**Pull request merged**